### PR TITLE
Add user profile, allow to edit it, if server supports it

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: app/src/main/res/drawable/ic_password.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (AndyScherzinger) [inline app/src/main/res/drawable/ic_password.xml:1]: licence header missing

File: app/src/main/res/drawable/upload.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (AndyScherzinger) [inline app/src/main/res/drawable/upload.xml:1]: licence header missing

**Comment by mahibi** [comment]
editing fields don't show up when clicking on "edit" if initially no field has any data

**Comment by mahibi** [comment]
- enter edit mode
- bring app to background / turn off display
- open app again
--> editing mode is active but button says "EDIT" instead of "SAVE"

**Comment by mahibi** [comment]
i see no avatar (also not the generated one with letters) and have no ability to set any

File: app/src/main/java/com/nextcloud/talk/controllers/ProfileController.java [hunk 3]
```diff
@@ -0,0 +1,800 @@
...
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(new Observer<GenericOverall>() {
+                            @Override
+                            public void onSubscribe(@NotNull Disposable d) {
+                            }
+
+                            @Override
+                            public void onNext(@NotNull GenericOverall userProfileOverall) {
+                                Log.d("ProfileController", "Successfully saved: " + item.text + " as " + item.field);
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/controllers/ProfileController.java:439]: use a static TAG for "ProfileController"

**Comment by mahibi** [comment]
- [ ] please describe changes in the changelog.md

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -346,14 +346,15 @@
 
     <!-- Phonebook Integration -->
     <string name="nc_settings_phone_book_integration_key" translatable="false">phone_book_integration</string>
-    <string name="nc_settings_phone_book_integration_desc">Match contacts based on phone number to integrate Talk shortcut in phone book</string>
-    <string name="nc_settings_phone_book_integration_title">Phone book integration</string>
+    <string name="nc_settings_phone_book_integration_desc">Match contacts based on phone number to integrate Talk
+        shortcut into system contacts app</string>
```
Comment (tobiasKaminsky) [inline app/src/main/res/values/strings.xml:350]: Line wraps might kill our translation system, so please avoid them.

**Comment by tobiasKaminsky** [comment]
> editing fields don't show up when clicking on "edit" if initially no field has any data

How did you managed to have this?
I cannot remove full name nor email, once it is set.

**Comment by tobiasKaminsky** [comment]
> > editing fields don't show up when clicking on "edit" if initially no field has any data
> 
> How did you managed to have this?
> I cannot remove full name nor email, once it is set.

New user it is…

**Comment by tobiasKaminsky** [comment]
> i see no avatar (also not the generated one with letters) and have no ability to set any

I guess that some server side PR is missing.

**Comment by tobiasKaminsky** [comment]
:exclamation: rebased :exclamation:

**Comment by mahibi** [comment]
22 new FindBugs warnings? @tobiasKaminsky

**Comment by AndyScherzinger** [comment]
❗ rebased ❗ 

+ plus added some improvements to the UI xml code

File: app/src/main/java/com/nextcloud/talk/controllers/ProfileController.java (hunk 3 above)
Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/controllers/ProfileController.java]: DONE

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @mahibi latest commits optimize layouting (whitespaces and dark/light theming)

**Review by AndyScherzinger — APPROVED** [review]

**Review by mahibi — APPROVED** [review]

**Pull request merged**